flink-runtime/src/main/java/org/apache/flink/runtime/jobmanager/scheduler/SlotSharingGroup.java

```diff
@@ -38,14 +40,20 @@
 
 	private final SlotSharingGroupId slotSharingGroupId = new SlotSharingGroupId();
 
+	/** Represents resources of all tasks in the group. Default to be zero.
+	 * Any task with UNKNOWN resources will turn it to be UNKNOWN. */
+	private ResourceSpec resourceSpec = ResourceSpec.newBuilder(0.0, 0).build();
+
 	// --------------------------------------------------------------------------------------------
 
-	public void addVertexToGroup(JobVertexID id) {
-		this.ids.add(id);
+	public void addVertexToGroup(final JobVertex jobVertex) {
```

Comment:
alternatively we could explicitly pass both the id and resource spec.
We usually create problems down the line if we spread classes like vertex classes.

Reply:
Agreed. Updated.